README.md

```diff
@@ -10,33 +10,74 @@
 dotnet add package KubernetesClient
 ```
 
-# Generating the Client Code
+# Usage
 
-## Prerequisites
+## Authentication/Configuration
+You should be able to use an standard KubeConfig file with this library,
+see the `BuildConfigFromConfigFile` function below. Most authentication
+methods are currently supported, but a few are not, see the 
+[known-issues](https://github.com/kubernetes-client/csharp#known-issues)
 
-You'll need a Linux machine with Docker.
+You should also be able to authenticate using the in-cluster service
+account using the `InClusterConfig` function shown below.
 
-The generated code works on all platforms supported by .NET or .NET Core.
+## Sample Code
 
-Check out the generator project into some other directory
-(henceforth `$GEN_DIR`)
+### Creating the client
+```c#
+// Load from the default kubeconfig on the machine.
+var config = KubernetesClientConfiguration.BuildConfigFromConfigFile();
 
-```bash
-cd $GEN_DIR/..
-git clone https://github.com/kubernetes-client/gen
+// Load from a specific file:
+var config = KubernetesClientConfiguration.BuildConfigFromConfigFile(Environment.GetEnvironmentVariable("KUBECONFIG"));
+
+// Load from in-cluster configuration:
+var config = KubernetesClientConfiguration.InClusterConfig()
+
+// Use the config object to create a client.
+var client = new Kubernetes(config);
 ```
 
-## Generating code
+### Listing Objects
+```c#
+var namespaces = client.ListNamespace();
```

Comment:
Nitpick: We should prefer showing the async methods to the blocking ones.  It is easy to translate in either direction of course, but we should set the expectation that async is the 'normal' way to do it.

Reply:
the examples are all sync right now. Let's do this in a separate PR.